### ISSUE TITLE
Unit Tests

### ISSUE THREAD

**Issue by Odonno** [Odonno/Modern-Gitter#36]
Implements Unit Tests, using Microsoft Visual Studio Test Framework.

**Comment by gep13** [comment]
From everything I have heard, using MSTest might not be the way to go, unless there is a compelling reason.  Something like NUnit or XUnit, might be the better way to go.

**Comment by Odonno** [comment]
Without reason, I will not choose other test framework.

**Comment by gep13** [comment]
This is from the guy who literally wrote the book about Unit Testing in .Net:

http://blog.ploeh.dk/2010/04/26/WhyImmigratingfromMSTesttoxUnit.net/

**Comment by Odonno** [comment]
Sure, I'll read that.

**Comment by Odonno** [comment]
I read the post, but it is very old. I think MSTest is really well implemented now, but clearly, the Fact/Theory of xUnit is intriguant. Why not going to xUnit this time ?

**Comment by gep13** [comment]
Agreed, it is an older article, but I think some of the points that are made are still valid.  If you search, you will likely find more up to date articles.

Sorry, not sure what you meant by your last question.

**Comment by Odonno** [comment]
Just a question to myself. :)
We'll go for xUnit when 1.3 is over.

**Comment by qujck** [comment]
I always preferred xUnit until I started working with Wpf. Wpf has to run in the STA and xUnit does not support this [in the core framework](https://github.com/xunit/xunit/issues/103). I don't know if the same restriction exists with Windows Phone apps but I'm letting you know just in case. [NUnit](http://nunit.org/) does support Sta with [`RequiresSTAAttribute` and `STAThreadAttribute`](http://www.nunit.org/index.php?p=requiresSTA&r=2.5)

P.S. [This guy](http://osherove.com/) wrote ***the*** book on [Unit Testing in .Net](http://artofunittesting.com/) ;-)

**Comment by Odonno** [comment]
Thanks @qujck,

I think we'll now soon if xUnit does not suite this project. For now, there is a lot of things to complete before starting Unit Tests, I need to free time. Not started in TDD is foolish, I know now. ^^

**Comment by gep13** [comment]
>@qujck said...
P.S. This guy wrote the book on Unit Testing in .Net ;-)

Doh! That is the book that I was thinking about, but I clearly got my authors mixed up :smile:   Mark wrote the Dependency Injection book.  Apologies.

**Comment by Odonno** [comment]
Starting with Unit Tests on ViewModels, I think it's the most important right now.
I am using xUnit.

**Comment by gep13** [comment]
Sounds like the right approach to me :+1:

**Comment by Odonno** [comment]
I have a done a large part of Unit Tests on ViewModels (only Windows 8.1 tests only).

I will update the first post to show where we are.
